pkg/load/agents/configAgent.go

```diff
@@ -35,8 +40,12 @@ type configAgent struct {
 	cycle        time.Duration
 	generation   int
 	errorMetrics *prometheus.CounterVec
+	indexFuncs   map[string]IndexFn
+	indexes      map[string]configIndex
 }
 
+type configIndex map[string][]*api.ReleaseBuildConfiguration
```

Comment:
By making these pointers I wonder if we'll run into mutability issues

Reply:
Even if we returned this by value we had the issues because it contains pointer fields. The thinking was that if we have that problem anyways, we might as well save us the allocation.